README.md

```diff
@@ -368,6 +367,7 @@ interface Observable {
   Observable take(unsigned long long);
   Observable drop(unsigned long long);
   Observable flatMap(Mapper mapper);
+  Observable finally(VoidFunction callback);
```

Comment:
Seems that https://github.com/WICG/observable/pull/49#discussion_r1324578891 has not been addressed yet.

Reply:
Updated.

Reply:
I think I'm still confused why this takes a callback and returns an Observable. That's what I'm pointing to above: https://github.com/WICG/observable/pull/49#discussion_r1324578891.

Reply:
Ah... this is a method for basically saying "When this observable finalizes, I want to do X (usually perform some clean up) every time" Because you don't know if it will error or complete or just be unsubscribed.

It composes to an observable, because it doesn't cause a subscription. If that makes sense.

```ts

document.on('mousedown').flatMap(() => {
  const div = document.createElement('div');
  document.body.append(div);

  return document.on('mousemove')
    .takeUntil(document.on('mouseup'))
    .finally(() => div.remove())
})
```

Reply:
> It composes to an observable, because it doesn't cause a subscription. If that makes sense.

This is pretty much what I've been asking about. If it doesn't cause subscription, *and* it passes in its handler function into the operator, then why does it need to compose to anything at all? Why can't it just return undefined. I feel like it's somewhat confusing for it to take in a callback function handler, _and_ also allow subscription handlers: which gets called first? Can anything inside the `finally()` callback function influence what `Observer` callback gets fired on the subscription?

Although in re-asking this, I guess I somewhat figured out the answer to my question: if it returns undefined, then you actually have no way of subscribing to it at all ever, so I guess that doesn't work at all :/